paimon-core/src/main/java/org/apache/paimon/io/KeyValueFileWriterFactory.java

```diff
@@ -142,14 +142,15 @@ private KeyValueDataFileWriter createDataFileWriter(
                         fileIndexOptions);
     }
 
-    public void deleteFile(String filename, int level) {
-        fileIO.deleteQuietly(formatContext.pathFactory(level).toPath(filename));
+    public void deleteFile(DataFileMeta meta, int level) {
+        fileIO.deleteQuietly(formatContext.pathFactory(level).toPath(meta));
     }
 
-    public void copyFile(String sourceFileName, String targetFileName, int level)
+    public void copyFile(
+            String sourceFileName, String targetFileName, int level, String externalPath)
```

Comment:
We can pass two data file metas to this method.

Reply:
done